app/src/auth/components/SessionProvider/index.jsx

```diff
@@ -16,23 +18,48 @@ type Props = {
 const storage = isLocalStorageAvailable() ? window.localStorage : null
 
 let cachedToken // fallback if no webstorage
+let cachedDate
+
+function hasExpired(date: Date) {
+    if (!date) { return true }
+
+    const diff = Math.abs(Date.now() - date.getTime())
```

Comment:
Shouldn't it be expired if `date - now < 6h`. With `abs` we seem to effectively have 12h window around "now" for date to not be expired. Example:

- now 5am and date 10am → diff: abs(5 - 10) → 5h → still valid
- now 10am and date 5am → diff: abs(10 - 5) → 5h → still valid